app/qml/ui/+nativemenubar/MenuBar.qml

```diff
@@ -358,6 +367,14 @@ Item {
 
             Platform.MenuSeparator {}
 
+            Platform.MenuItem {
+                objectName: "showBrushPreview"
+                text: qsTr("Show Brush Preview")
+                checkable: true
+                checked: settings.brushPreviewVisible
+                onCheckedChanged: settings.brushPreviewVisible = checked
```

Comment:
Please use onTriggered, as triggered() is emitted only when the user clicks the menu item, whereas checkedChanged() is emitted when it changes programmatically as well:

https://doc.qt.io/qt-5/qml-qt-labs-platform-menuitem.html#triggered-signal